### PR TITLE
8262883: doccheck: Broken links in java.base

### PR DESCRIPTION
**Pull request by Michael-Mc-Mahon**
Hi,

Could I get the following trivial doc changes reviewed please, caused by:

- broken <A> tags in MethodHandles referring to package.html instead of package-summary.html

- references to a package level #unixdomain anchor that no longer exists.

- a <a> tag missing a "../" in SocketChannel

Thanks,

Michael

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8262883](https://bugs.openjdk.java.net/browse/JDK-8262883): doccheck: Broken links in java.base

### Reviewers
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/3491/head:pull/3491` \
`$ git checkout pull/3491`

Update a local copy of the PR: \
`$ git checkout pull/3491` \
`$ git pull https://git.openjdk.java.net/jdk pull/3491/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 3491`

View PR using the GUI difftool: \
`$ git pr show -t 3491`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/3491.diff">https://git.openjdk.java.net/jdk/pull/3491.diff</a>

</details>

### CODE REVIEW

**Review by LanceAndersen — APPROVED** [review]

File: src/java.base/share/classes/java/nio/channels/DatagramChannel.java [hunk 1]
```diff
@@ -172,7 +172,7 @@ public static DatagramChannel open() throws IOException {
      * java.nio.channels.spi.SelectorProvider} object.  The channel will not be
      * connected.
      *
-     * @apiNote <a href="package-summary.html#unixdomain">Unix domain</a> sockets
+     * @apiNote {@linkplain java.net.UnixDomainSocketAddress Unix domain} sockets
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/channels/DatagramChannel.java:175]: The parameter to this method is a ProtocolFamily and maybe this note should be say that StandardProtocolFamily#UNIX is not supported.

**Comment by RogerRiggs** [comment]
Integrating less than an hour after the PR is started seems to be a bit too quick to allow useful input.
Perhaps even for 'trivial' changes.

File: src/java.base/share/classes/java/nio/channels/DatagramChannel.java (hunk 1 above)
Comment (Michael-Mc-Mahon) [reply src/java.base/share/classes/java/nio/channels/DatagramChannel.java:175]: Okay, reasonable suggestion. I was a bit quick integrating. I will file a new issue to update the spec as suggested.

**Pull request closed without merging**